launch_testing/launch_testing/__init__.py

```diff
@@ -0,0 +1,61 @@
+import io
+import re
+from launch.output_handler import LineOutput
+
+
+class InMemoryHandler(LineOutput):
+
+    def __init__(
+        self, name, ld, expected_lines, regex_match=False,
+        filtered_prefixes=[
+            b'pid', b'rc',
+            b'RTI Data Distribution Service Evaluation License issued to',
+            b'Expires on']):
+        super(LineOutput, self).__init__()
+        self.name = name
+        self.ld = ld
+        self.expected_lines = expected_lines
+        self.killed = False
+        self.left_over_stdout = b''
+        self.left_over_stderr = b''
+        self.stdout_data = io.BytesIO()
+        self.stderr_data = io.BytesIO()
+        self.regex_match = regex_match
+        self.matched = False
+        self.filtered_prefixes = filtered_prefixes
+
+    def on_stdout_lines(self, lines):
+        if self.killed:
+            return
+
+        for line in lines.splitlines():
+            # Filter out stdout that comes from underlying DDS implementations
+            if any([line.startswith(prefix) for prefix in self.filtered_prefixes]):
+                continue
+            self.stdout_data.write(line + b'\n')
+
+            # Are we ready to quit?
+            if self.regex_match:
+                self.matched = re.match(self.expected_lines[0], line)
+            else:
+                output_lines = self.stdout_data.getvalue().splitlines()
+                self.matched = output_lines == self.expected_lines
```

Comment:
Then this should at least be documented since the current combination of "rules" is only obvious to the person who has implemented the code. Just from the PR using this I would have expected something very different.